src/index.js

```diff
@@ -185,7 +185,115 @@ const getEncodingIdx = (type) => {
   }
 };
 
+/**
+ * @typedef {[number, number, number, number] | [number, number, number]} Color
+ *
+ * @typedef {'category' | 'value1' | 'valueA' | 'valueZ' | 'z'} Category
+ *
+ * @typedef {'value' | 'value2' | 'valueB' | 'valueW' | 'w'} Value
+ *
+ * @typedef {Category | Value} DataEncoding
+ *
+ * @typedef {DataEncoding | 'inherit' | 'segment'} PointDataEncoding
+ *
+ * @typedef {'alt' | 'cmd' | 'ctrl' | 'meta' | 'shift'} Key
+ *
+ * @typedef {'lasso' | 'rotate' | 'merge' | 'panZoom'} Action
+ *
+ * @typedef {'deselect' | 'lassoEnd'} LassoClearEvent
+ *
+ * @typedef {Record<string, any>} Camera2D
+ *
+ * @typedef {{
+ *   backgroundColor: Color | string;
+ *   backgroundImage: import('regl').Texture2D | string;
+ *   colorBy: DataEncoding;
+ *   deselectOnDblClick: boolean;
+ *   deselectOnEscape: boolean;
+ *   lassoColor: Color | string;
+ *   lassoLineWidth: number;
+ *   lassoMinDelay: number;
+ *   lassoMinDist: number;
+ *   lassoClearEvent: LassoClearEvent;
+ *   lassoInitiator: boolean;
+ *   keyMap: Record<Key, Omit<Action, 'panZoom'>>;
+ *   mouseMode: Omit<Action, 'merge'>;
+ *   showReticle: boolean;
+ *   reticleColor: Color | string;
+ *   pointColor: Color;
+ *   pointColorActive: Color;
+ *   pointColorHover: Color;
+ *   showPointConnections: boolean;
+ *   pointConnectionColor: Color;
+ *   pointConnectionColorActive: Color;
+ *   pointConnectionColorHover: Color;
+ *   pointConnectionColorBy: PointDataEncoding;
+ *   pointConnectionOpacity: number;
+ *   pointConnectionOpacityBy: PointDataEncoding;
+ *   pointConnectionOpacityActive: number;
+ *   pointConnectionSize: number;
+ *   pointConnectionSizeActive: number;
+ *   pointConnectionSizeBy: PointDataEncoding;
+ *   pointConnectionMaxIntPointsPerSegment: number;
+ *   pointConnectionTolerance: number;
+ *   pointSize: number;
+ *   pointSizeSelected: number;
+ *   pointSizeMouseDetection: 'auto';
+ *   pointOutlineWidth: number;
+ *   opacity: number;
+ *   opacityBy: DataEncoding;
+ *   opacityByDensityFill: number;
+ *   sizeBy: DataEncoding;
+ *   height: 'auto' | number;
+ *   width: 'auto' | number;
+ *   gamma: number;
+ * }} Settable
+ *
+ * @typedef {{
+ *   canvas: HTMLElement;
+ *   regl: import('regl').Regl;
+ *   syncEvents: boolean;
+ *   version: string;
+ *   lassoInitiatorElement: HTMLElement;
+ *   camera: Camera2D;
+ *   performanceMode: boolean;
+ *   opacityByDensityDebounceTime: number;
+ * } & Settable} Properties
+ *
+ * @param {Partial<Properties>} initialProperties
+ */
```

Comment:
Alternatively all types could be declared in `src/types.d.ts` and then imported in `src/index.js`:

```javascript
 *
 * @param {Partial<import('./types').Properties>} initialProperties
 */
```